usr/src/uts/common/os/log_sysevent.c

```diff
@@ -641,6 +642,10 @@ sysevent_add_attr(sysevent_attr_list_t **ev_attr_list, char *name,
 		error = nvlist_add_hrtime(*ev_attr_list, name,
 		    se_value->value.sv_time);
 		break;
+	case SE_DATA_TYPE_NVLIST:
```

Comment:
This will cause problems on FreeBSD, where sysevents are reported in a textual format as a list of key=value pairs.  But it's not too bad as long as the nvlist doesn't contain other nvlists.  Still, my preference would be to iterate over the list up in spa_history_log_notify.